### DISCUSSION TITLE
Tasmota Ir esp8266

### DISCUSSION THREAD

**Discussion by delaineS** [arendst/Tasmota#21196] [Support request] [answered]
Hi everyone, I'm trying to use an esp8266 to control an air conditioner (Stelbi brand).
I loaded the TasmotaIr firmware and configured it as in figure 1.
I followed this scheme https://github.com/hristo-atanasov/Tasmota-IRHVAC/blob/master/images/schematics.jpeg and the procedure I do is go to the console and send the command with the air conditioner remote control giving me this:

19:49:31.514 MQT: tele/Tasmota/RESULT = {"IrReceived":{"Protocol":"COOLIX","Bits":24,"Data":"0xB27BE0","DataLSB":"0x4DDE07"," Repeat":0,"IRHVAC":{"Vendor":"COOLIX","Model":-1,"Command":"Control","Mode":"Off","Power":"Off"," Celsius":"On","Temp":25,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":" Off","Econo":"Off","Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1, "iFeel":"Off","SensorTemp":null}}}

Then I send the same command:
19:51:19.567 CMD: irsend {"Protocol":"COOLIX","Bits":24,"Data":"0xB27BE0","DataLSB":"0x4DDE07","Repeat":0,"IRHVAC":{ "Vendor":"COOLIX","Model":-1,"Command":"Control","Mode":"Off","Power":"Off","Celsius":"On","Temp" :25,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off" ,"Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1,"iFeel":"Off","SensorTemp ":null}}

Response from the console:
19:51:19.892 MQT: stat/Tasmota/RESULT = {"IRSend":"Done"}

But nothing happens to the air conditioner.
What am I doing wrong?
I leave attached the photo of my scheme in figure 2![image](https://github.com/arendst/Tasmota/assets/92988734/36cc87c4-f5ef-4eef-a348-f00694d0fbaa)![image](https://github.com/arendst/Tasmota/assets/92988734/9591d942-864d-495a-afc1-cd291739ee2f)

**Comment by s-hadinger** [comment, marked as answer]
Please read the doc carefully. Your syntax does not exist https://tasmota.github.io/docs/Tasmota-IR/#sending-ir-commands

The correct is:
`IrHVAC { "Vendor":"COOLIX","Model":-1,"Command":"Control","Mode":"Off","Power":"Off","Celsius":"On","Temp" :25,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off" ,"Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1,"iFeel":"Off"}`

**Reply by delaineS** [reply]
it still doesnt work:

12:55:46.288 CMD: IRHVAC {"Vendor":"COOLIX","Model":-1,"Command":"Control","Mode":"Auto","Power":"On","Celsius":"On","Temp":27,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off","Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1,"iFeel":"Off"}
12:55:46.607 MQT: stat/Tasmota/RESULT = {"IRHVAC":{"Vendor":"COOLIX","Model":-1,"Command":"Control","Mode":"Auto","Power":"On","Celsius":"On","Temp":27,"FanSpeed":"Auto","SwingV":"Off","SwingH":"Off","Quiet":"Off","Turbo":"Off","Econo":"Off","Light":"Off","Filter":"Off","Clean":"Off","Beep":"Off","Sleep":-1,"iFeel":"Off","SensorTemp":null}}

**Reply by s-hadinger** [reply]
I'm afraid we can't help you from here. The encoding is manager by IRRemoteESP8266 library. Please check on their repository for protocol troubleshooting.

**Comment by NsinghP** [comment]
Try using Raw data in your IRSend command, first set your device to capture raw data using the SetOption58 command `SetOption58 1` , this will enable you to capture Raw IR Codes. Then extract the Raw IR code data and incorporate it into your IRSend command, below is a sample of command I used to turn my Air conditioner ON:
`IRSend 0,+3510-1615+570-1130+545-1150C-490E-515EhE-1155EhCgCdCdEhEfEhCgEiEiEh+550fEiE-510EhEiEkEhEiEkEk+575gCgJkEh+540hCgEhCgEhEhEhEkCgEhEkEiEkEkCdEhMhEkEhEkEiEhEhEhJkC-1125CdEfCdCgEhE-520MhEhEhEkEiEiEiEhC-485EkEhEhCgEhEhCgCgEhEhEkEhEhEkEhEkEhEhEhCgMhEhCpEhCgEhEhEhEhMhCgCgEhMhMhEiEhEhEhEiE`

I hope this works for you, it worked very well in my case. See post [#20958](https://github.com/arendst/Tasmota/discussions/20958)

**Reply by delaineS** [reply]
Thank you for your reply! I tried with raw but still doesnt work!! Im Thinking to say goodbye to idea!
